apps/application-system/api/src/app/modules/application/e2e/payment/payment.service.spec.ts

```diff
@@ -177,9 +199,31 @@ describe('Payment Service', () => {
     expect(result.fulfilled).toBe(true)
   })
 
+  it('should find charge items using quantity', async () => {
```

Comment:
Consider adding a check for whether each of the charge items has the correct quantity in addition to checking the sum of charge items.